Userland/Libraries/LibGfx/Color.h

```diff
@@ -151,10 +151,10 @@ class Color {
 
     Color interpolate(const Color& other, float weight) const noexcept
     {
-        u8 r = red() + roundf(static_cast<float>(other.red() - red()) * weight);
-        u8 g = green() + roundf(static_cast<float>(other.green() - green()) * weight);
-        u8 b = blue() + roundf(static_cast<float>(other.blue() - blue()) * weight);
-        u8 a = alpha() + roundf(static_cast<float>(other.alpha() - alpha()) * weight);
+        u8 r = red() + (AK::round_to_int<i16>(static_cast<float>(other.red() - red()) * weight) & 0xff);
+        u8 g = green() + (AK::round_to_int<i16>(static_cast<float>(other.green() - green()) * weight) & 0xff);
+        u8 b = blue() + (AK::round_to_int<i16>(static_cast<float>(other.blue() - blue()) * weight) & 0xff);
+        u8 a = alpha() + (AK::round_to_int<i16>(static_cast<float>(other.alpha() - alpha()) * weight) & 0xff);
```

Comment:
I doubt wrapping is the desired behavior, while it follows the original behavior

Reply:
Might be nice to assert before the actual body of the function that 0 <= weight <= 1.